master/internal/task/allocation.go

```diff
@@ -553,6 +553,10 @@ func (a *Allocation) ResourcesStateChanged(
 	ctx.Log().Debugf("resources state changed: %+v", msg)
 	switch msg.ResourcesState {
 	case sproto.Pulling:
+		if !model.LessAllocationState(a.getModelState(), model.AllocationStatePulling) {
```

Comment:
this is basically identical to the change on line 575

Reply:
Interesting.   This stuff only gets set on Pulling?   Before Rig added that state relatively recently, we went directly to Running.    There is actually no guarantee that we will see a Pulling state -- it depends upon timing.
Is that OK?

Reply:
oh, goodness. ok, we should chat.

Reply:
ok, fixed it to handle this